packages/orm/src/collections/ManyToManyCollection.ts

```diff
@@ -150,7 +150,7 @@ export class ManyToManyCollection<T extends Entity, U extends Entity> extends Ab
 
   /** Some random entity got deleted, if it was in our collection, remove it. */
   onDeleteOfMaybeOtherEntity(maybeOther: Entity): void {
-    ensureNotDeleted(this.entity);
+    ensureNotDeleted(this.entity, { ignore: "pending" });
```

Comment:
Another data point for `ensureNotDeleted` should either always ignore pending or just go away...